src/praisonai/praisonai/bots/_delivery_control_store.py

```
@@ -0,0 +1,355 @@
+"""
+Shared SQLite store for cross-worker delivery-control state (issue #2579).
+
+The durable delivery stack (outbox, DLQs, ingress journal, approval store) is
+already SQLite-first. The *delivery-control* state that decides how fast and
+whether to send, however, was per-process: the rate limiter kept its token
+bucket in memory and the dead-target registry rewrote a whole JSON file on
+every change. Under a horizontally-scaled gateway N workers each keep their own
+token bucket, so the global send rate becomes ~N× the platform ceiling — exactly
```

Ruff (0.15.20)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)